src/libraries/System.Private.CoreLib/src/System/Tuple.cs

```diff
@@ -359,26 +380,37 @@ int IStructuralComparable.CompareTo(object? other, IComparer comparer)
 
         public override int GetHashCode()
         {
-            return ((IStructuralEquatable)this).GetHashCode(EqualityComparer<object>.Default);
+            return GetHashCodeCommon(EqualityComparer<object>.Default);
         }
 
         int IStructuralEquatable.GetHashCode(IEqualityComparer comparer)
         {
-            return Tuple.CombineHashCodes(comparer.GetHashCode(m_Item1!), comparer.GetHashCode(m_Item2!), comparer.GetHashCode(m_Item3!));
+            return GetHashCodeCommon(comparer);
         }
 
         int ITupleInternal.GetHashCode(IEqualityComparer comparer)
         {
-            return ((IStructuralEquatable)this).GetHashCode(comparer);
+            return GetHashCodeCommon(comparer);
+        }
+
+        private int GetHashCodeCommon(IEqualityComparer comparer)
```

Comment:
Just wondering - why does this need `Common` suffix vs. everything else works fine without it?

Reply:
Just leftover from the first version. I removed the suffix.